w3af/core/data/db/variant_db.py

```diff
@@ -201,6 +209,32 @@ def _seen_exactly_the_same(self, fuzzable_request, request_hash):
         self._log_return_false(fuzzable_request, 'seen_exactly_the_same')
         return False
 
+    def _request_is_looped(self, fuzzable_request):
+        """
+        Check if the current request is the looped one.
+        If the requested URL seems like:
+        http://example.com/redirect/redirect/redirect/redirect/redirect/index.html
+        that means crawler probably fallen into infinite crawling loop due to `replace()`
+        in JavaScript history API.
+
+        :param FuzzableRequest fuzzable_request: request we're currently scanning
+        :returns Boolean: True if request is looped.
+        """
+        url = fuzzable_request.get_uri()
+        path, head = os.path.split(url.path)
+        same_path_parts = 0
+        # first iteration will almost never increment same_path_parts, because
+        # first head will be probably something like 'index.html'
+        for _ in range(self.path_max_loop + 1):
+            prev_head = head
+            path, head = os.path.split(path)
+            if head and head == prev_head:
+                same_path_parts += 1
+
+        if same_path_parts >= self.path_max_loop:
```

Comment:
I freaking love your solution. It is amazingly simple!

Reply:
I'm only afraid that returning `False` in `VariantDB.append()` will finish whole scan before web_spider will crawl the rest of URLs. Do you know if it will work this way?

Reply:
The scan will not stop if you return false in `append()`. The only action that should be done after `False` is returned is to ignore that "looped URL".